app/src/ui/create-branch/create-branch-dialog.tsx

```diff
@@ -143,61 +155,40 @@ export class CreateBranch extends React.Component<
         </p>
       )
     } else if (tip.kind === TipState.Valid) {
-      const currentBranch = tip.branch
+      if (
+        this.props.upstreamGitHubRepository !== null &&
+        this.props.upstreamDefaultBranch !== null
+      ) {
+        return this.renderForkBranchSelection(
+          tip.branch.name,
+          this.props.upstreamDefaultBranch,
+          this.props.upstreamGitHubRepository.fullName
+        )
+      }
+
       const defaultBranch = this.state.isCreatingBranch
-        ? this.props.defaultBranch
+        ? getBranchForStartPoint(this.state.startPoint, this.props)
```

Comment:
We would only get here in the non-fork case right? I'm wondering if we would ever expect to **not** have `this.props.defaultBranch` as the value here.

I'm wondering how sound this logic is: for the non-fork case, `this.state.startPoint` would be `StartPoint.DefaultBranch`(since `props.upstreamDefaultBranch` would be null), and we would expect the return value from `getBranchForStartPoint` to be `this.props.defaultBranch` (with the change I suggested below 👇) 

@outofambit does this all sound reasonable or am I missing something?

Reply:
yes that is all correct @kuychaco! with your fix below, this should always return `this.props.defaultBranch`, but i do see your point that that extra logic is not really necessary here since we already know we're in the non-fork case here. therefore, the original logic should work here. amending now.

```suggestion
        ? this.props.defaultBranch
```